cmd/setup.go

```diff
@@ -136,21 +134,25 @@ func (o *SetupOptions) DoSetup(f Factory) (err error) {
 
 		err = json.Unmarshal([]byte(o.ConfigData), cfg)
 		if cfg.Profile != nil {
-			o.Peername = cfg.Profile.Peername
+			o.Username = cfg.Profile.Peername
 		}
 		if err != nil {
 			return err
 		}
 	}
 
-	if o.Peername != "" {
-		cfg.Profile.Peername = o.Peername
-	} else if cfg.Profile.Peername == doggos.DoggoNick(cfg.Profile.ID) && !o.Anonymous {
-		cfg.Profile.Peername = inputText(o.Out, o.In, "choose a peername:", doggos.DoggoNick(cfg.Profile.ID))
+	// Handle the --username flag, or prompt the user for a username
+	if o.Username != "" {
+		cfg.Profile.Peername = o.Username
+	} else if !o.Anonymous {
```

Comment:
we should also obey `noPrompt` here.

Reply:
`noPrompt` is not supported by `setup`, because the function that assigns `noPrompt` is Qri.Init which runs when an instance is created. An instance isn't created for `setup` because it can't be created before the repo exists. Should we add `noPrompt` handling here that copies what Qri.Init is doing? If so, is there any point in having both `--no-prompt` and `--anonymous`?

Reply:
ah that makes sense. We also have the `--setup` flag itself, which is a no-prompt setup. Ok my bad ignore me, Let's merge!